core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -227,11 +227,18 @@ class CityInfo {
     }
 
     fun reassignWorkers() {
-        workedTiles = hashSetOf()
-        population.specialists.clear()
-        for (i in 0..population.population)
-            population.autoAssignPopulation()
-        cityStats.update()
+        var foodWeight = 1f
+        var foodPerTurn = 0f
+        while (foodWeight < 3 && foodPerTurn <= 0) {
```

Comment:
So foodWeight increments from 1 to 3 by 0.f increments, and at each once checks whether the foodPerTurn is not negative? This is good for larger cities, but for smaller cities (say, pop < 5) we want them to grow, not to remain stagnant... do they still grow well with the in place?

Reply:
This feature tries to get minimum food net gain. I think it also applies to small cities because if a city has negative food, pop will die out hindering its production. Your suggestions?

Reply:
Or stuck in small pop, limiting its potential.

Reply:
Oh, sorry I misunderstood your question. foodWeight parameter is multiplied by previous evaluation, so actual foodWeight is at least 1.2.

Reply:
Cool =)